packages/mui-material/src/Tooltip/Tooltip.js

```diff
@@ -594,15 +595,17 @@ const Tooltip = React.forwardRef(function Tooltip(inProps, ref) {
       },
     ];
 
-    if (PopperProps.popperOptions?.modifiers) {
-      tooltipModifiers = tooltipModifiers.concat(PopperProps.popperOptions.modifiers);
+    if (slotProps.popper?.popperOptions?.modifiers || PopperProps?.popperOptions?.modifiers) {
```

Comment:
For me this does not work because `slotProps` might be a function that receives `ownerState`
So we need to resolve the slot props before this condition to have the object version always accessible

Reply:
@flaviendelangle you are right. But with the changes it should work now.